backend/hitas/models/codes.py

```diff
@@ -0,0 +1,73 @@
+import re
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from hitas.models._base import HitasModel
+
+
+def validate_code_number(value: str) -> None:
+    # Example valid value: '012'
+    match = re.search(r"^\d{3}$", value)
+
+    if match is None:
+        raise ValidationError(
+            _("%(value)s is not an valid code number"),
+            params={"value": value},
+        )
+
+
+class AbstractCode(HitasModel):
+    value = models.CharField(max_length=1024)
+    description = models.TextField(null=True, blank=True)
+
+    in_use = models.BooleanField(default=True)
+    order = models.PositiveSmallIntegerField(null=True, blank=True)
+    legacy_code_number = models.CharField(max_length=3, unique=True, validators=[validate_code_number])
+    legacy_start_date = models.DateTimeField(default=timezone.now)
+    legacy_end_date = models.DateTimeField(null=True, blank=True)
```

Comment:
What's the point of legacy dates? 🤔

Reply:
We might not need them - if we do, we can rename